app/scripts/metamask-controller.js

```diff
@@ -809,6 +809,19 @@ export default class MetamaskController extends EventEmitter {
       messenger: currencyRateMessenger,
       state: initState.CurrencyController,
     });
+    const initialFetchExchangeRate =
+      this.currencyRateController.fetchExchangeRate.bind(
+        this.currencyRateController,
+      );
+    this.currencyRateController.fetchExchangeRate = (...args) => {
+      if (this.preferencesController.store.getState().useCurrencyRateCheck) {
+        return initialFetchExchangeRate(...args);
+      }
+      return {
+        conversionRate: 0,
+        usdConversionRate: 0,
```

Comment:
```suggestion
        conversionRate: null,
        usdConversionRate: null,
```

Can we handle non-numeric values? Anything to allow differentiating rates not being available from actual `0` would be useful?

`null`/`undefined`/`NaN`.